internal/daemon/config.go

```diff
@@ -31,62 +37,62 @@ func (c *ConfigFile) SetDefaults() {
 // Assert interface compliance.
 var _ defaults.Setter = (*ConfigFile)(nil)
 
-// config holds the configuration state as a singleton. It is used from LoadConfig and Config
-var config *ConfigFile
-
-// LoadConfig loads the daemon config from given path. Call it only once when starting the daemon.
-func LoadConfig(path string) error {
-	if config != nil {
-		return errors.New("config already set")
+func (c *ConfigFile) Validate() error {
```

Comment:
While changing this, maybe add a one-line API documentation stating that this method validates the whole configuration?

Reply:
I have also added which interface it implements.